backend/src/main/java/gov/cdc/usds/simplereport/utils/BulkUploadResultsToFhir.java

```diff
@@ -210,36 +232,63 @@ private Bundle convertRowToFhirBundle(TestResultRow row, UUID orgId) {
             row.getOrderingProviderId().value,
             new PersonName(
                 row.getOrderingProviderFirstName().value,
-                row.getOrderingProviderLastName().value,
                 row.getOrderingProviderMiddleName().value,
+                row.getOrderingProviderLastName().value,
                 null),
             row.getOrderingProviderPhoneNumber().value,
             providerAddr,
             DEFAULT_COUNTRY,
             row.getOrderingProviderId().value);
 
-    var additionalDeviceData =
-        deviceTypeRepository.findDeviceTypeByModelIgnoreCase(row.getEquipmentModelName().value);
-
     String equipmentUid = null;
     String testKitNameId = null;
     String manufacturer = null;
+    String diseaseName = null;
     UUID deviceId = UUID.randomUUID();
-    if (additionalDeviceData != null) {
-      var diseaseCode = row.getTestPerformedCode().value;
+    var testPerformedCode = row.getTestPerformedCode().value;
+    var modelName = row.getEquipmentModelName().value;
+    var matchingDevices =
+        availableDevices.stream()
+            .filter(device -> device.getModel().equalsIgnoreCase(modelName))
+            .filter(
+                device ->
+                    device.getSupportedDiseaseTestPerformed().stream()
+                        .anyMatch(
+                            disease ->
+                                Objects.equals(
+                                    disease.getTestPerformedLoincCode(), testPerformedCode)))
+            .toList();
```

Comment:
This can be optimized by pre-building a hashmap of the devices and using the device model+testPerformedCode as the key.